### PR TITLE
[hpx] Update to 1.8.1

### LINKED ISSUE #30306: [hpx] Update to version 1.8.1
Library name: hpx

New version number: 1.8.1

Other information that may be useful (release notes, etc...)

https://github.com/STEllAR-GROUP/hpx/releases/tag/1.8.1

### PR DESCRIPTION
**Pull request by bzindovic**
Fixes #30306.

### CODE REVIEW

File: ports/hpx/portfile.cmake [hunk 1]
```diff
@@ -6,13 +6,11 @@ string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "static" HPX_WITH_STATIC_LINKING
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO STEllAR-GROUP/hpx
-    REF f48c81865800ae72618b40b4e79d4168cfb0bd56
-    SHA512 06d1384615b327194d871145c1899317bd350c43a072be2cbfdc773f7869b71aafce3e9cabf835a8fe902a13d7050d5e0400a76f74023985575347a645196b1d
+    REF c6124b0
```
Comment (LilyWangLL) [inline ports/hpx/portfile.cmake:9]:
```suggestion
    REF "${VERSION}"
```

File: ports/hpx/vcpkg.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "hpx",
-  "version": "1.8.0",
-  "port-version": 2,
+  "version": "1.8.1",
+  "port-version": 0,
```
Comment (LilyWangLL) [inline ports/hpx/vcpkg.json:4]:
```suggestion
```

**Review by LilyWangLL — CHANGES_REQUESTED** [review]

File: ports/hpx/portfile.cmake [hunk 3]
```diff
@@ -6,13 +6,11 @@ string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "static" HPX_WITH_STATIC_LINKING
...
+    SHA512 82385a2e1f83f90946ec84c4a82ce9bfd1ae6495f556488d9f4095134e2131145f10bcb44abc755efedc76cc92e34b8115dbf5989e06c9096edecb9a2aeecc28
```
Comment (LilyWangLL) [inline ports/hpx/portfile.cmake:10]:
```suggestion
    REF "${VERSION}"
    SHA512 af2471a04dd0a3c414907ed06661ab1c6f6a49cc09d1ed3ae5d5587ca365270797a1d2ce9d0320dc7d7f9ff2c6d29037c7fbb84fa6d9c0033628ba7036f12986
```

**Review by LilyWangLL — CHANGES_REQUESTED** [review]

**Comment by LilyWangLL** [comment]
All features passed with following triplets:
```
x86-windows
x64-windows
```

**Comment by bzindovic** [comment]
@LilyWangLL I've added you suggestions to the fork. Should I mark that the this package with all the features is windows-only (original package can be built on Linux/Mac)?

**Comment by LilyWangLL** [comment]
> @LilyWangLL I've added you suggestions to the fork. Should I mark that the this package with all the features is windows-only (original package can be built on Linux/Mac)?

This is not needed, I just tested the features on windows. You should refer to the following steps to update version files of `hpx`.
```
##[error]versions/h-/hpx.json(2,): error : while parsing versions for hpx from version/h-/hpx.json
Version 1.8.1 was not found in versions file.
Run:
vcpkg x-add-version hpx
to add the new port version.
To attempt to resolve all errors at once, run:
vcpkg x-add-version --all
```

**Review by LilyWangLL — APPROVED** [review]

**Pull request merged**